contrib/dracut/Makefile.am

```diff
@@ -1,3 +1,4 @@
-SUBDIRS = 02zfsexpandknowledge 90zfs
+SUBDIRS = 02zfsexpandknowledge 90zfs $(ZFS_INIT_SYSTEMD)
+DIST_SUBDIRS = 02zfsexpandknowledge 90zfs systemd
```

Comment:
why did you separate it into a non-numbered directory?

Reply:
Since this systemd service file is dracut specific and I would expect it to be installed with the zfs-dracut package, it seemed logic to me to move it to the dracut directory. Maybe this should be even moved to `90zfs`?

Reply:
Since this systemd service file is dracut specific and I would expect it to be installed with the zfs-dracut package, it seemed logic to me to move it to the dracut directory. Maybe this should be moved to `90zfs`?

Reply:
why not `etc/system/systemd`?

Reply:
`/etc/systemd/system` is for the system administrator. Packages should install to `/usr/lib/systemd/system` or `/lib/systemd/system`, depending on the distro. The correct path can be obtained via `pkg-config`.

Reply:
@rlaager kpande meant `etc/systemd/system` in the repo which gets installed to `@systemdunitpath@`

@kpande As I said, this is dracut specific. `etc/systemd/system` are general zfs units. Why is the generator in `contrib/dracut` when it is systemd?

Reply:
@rlaager then maybe [this](https://github.com/zfsonlinux/zfs/tree/master/etc/systemd/system) should not be here?

Reply:
@c0d3z3r0 I've actually got no idea why there is a systemd generator in `contrib/dracut`.

Reply:
@kpande Great :D   I'd be ok with moving the unit file to the systemd tree, while keeping it installed only with `zfs-dracut`. I just want to see if @behlendorf has some thoughts on this.

Reply:
maybe @prometheanfire can chime in since i think he is the one who created that generator

Reply:
I could have easily created it wrong.  I don't create them every day.  It certainly sounds wrong if it's trying to run systemctl things on non-systemd systems though.

Reply:
ok, if we can i'd like to keep systemd generators in one place, as it may not end up being dracut specific. maybe @c0d3z3r0 can move them both into `etc/systemd` or as @rlaager suggested maybe that also needs its own issue to then move everything into the path provided by the system.

Reply:
> ok, if we can i'd like to keep systemd generators in one place, as it may not end up being dracut specific. maybe @c0d3z3r0 can move them both into `etc/systemd` 

Ok

> or as @rlaager suggested maybe that also needs its own issue to then move everything into the path provided by the system.

No, that is not needed since it is only the tree structure in git which is misleading. The files do end up in `/usr/lib/....` or `/lib/....` based on `configure`'s arguments

Reply:
Hm. That generator is only called from dracut. It is directly put to the initramfs without ever being called on a fully booted system.... https://github.com/zfsonlinux/zfs/blob/master/contrib/dracut/90zfs/module-setup.sh.in#L67

I'm not sure if it is a good idea to put a generator to  `/lib`/`usr/lib` that is never called, except from within dracut.

Reply:
ok, I would like for us to find some path for dracut generators other than the module directory. it is not immediately necessary so long as all of this works on dracut+openrc|systemd.

Reply:
I looked a bit through other dracut modules and found this one for example: [nbd](https://git.kernel.org/pub/scm/boot/dracut/dracut.git/tree/modules.d/95nbd)
The generator is in the module directory, so 90zfs should be the right place for both the generator and the unit file

Reply:
Here is another one https://git.kernel.org/pub/scm/boot/dracut/dracut.git/tree/modules.d/98dracut-systemd/emergency.service

Reply:
I just pushed my changes. IMHO that is the right way :-)